src/lib/cmp.js

```diff
@@ -0,0 +1,31 @@
+// This file supports both TCF v1 (__cmp) and TCF v2 (__tcfapi).
+// Remove support for TCF v1 once it is no longer supported by the community.
+//
+// Functions in this file should register the callback with both __cmp and __tcfapi, if possible.
+// The loaded CMP must only implement one of the two functions, ensuring the callback is only called once.
+import log from './log';
+
+export const getTCData = (view, callback) => {
+	if (view.__tcfapi === undefined && view.__cmp === undefined) {
+		return log.error('__tcfapi and __cmp are both undefined');
+	}
+
+	if (view.__tcfapi !== undefined) {
+		view.__tcfapi('addEventListener', 2, (data, addSuccess) => {
+			if (addSuccess && data.eventStatus === 'useractioncomplete') {
```

Comment:
This was based on the Quantcast documentation and does not take other CMPs into consideration.

Quantcast only returns a [TCData](https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/TCFv2/IAB%20Tech%20Lab%20-%20CMP%20API%20v2.md#tcdata) object with a non-empty `tcString` if the `eventStatus === 'useractioncomplete`. The callback is always invoked with this status even if the user has already given consent. The `tcString` is always empty when the callback is invoked with `eventStatus === 'tcloaded'`.

Cookiebot, however, only invokes the callback with `eventStatus === 'useractioncomlete'` after a user has given consent. If a user has already given consent then the callback is invoked with `eventStatus === 'tcloaded'` when then contains a non-empty `tcString`.

We should change this condition such that it checks if `tcString` is not empty, instead of checking the `eventStatus`.
All documentation should be updated accordingly.

Links:
- [addEventListener](https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/TCFv2/IAB%20Tech%20Lab%20-%20CMP%20API%20v2.md#addeventlistener)